portal-kernel/src/com/liferay/portal/kernel/upgrade/UpgradeProcess.java

```diff
@@ -115,6 +142,17 @@ public void upgrade() throws UpgradeException {
 			throw new UpgradeException(throwable);
 		}
 		finally {
+			if (connection != null) {
```

Comment:
@LuisOrtiz89 why is there a null check here? connection is never null in this method.